File: frontend/src/app/model/projectmodel.ts
import {UserModel} from "./usermodel";

export class ProjectModel {
  id: string;
  projectcode: string;
  projectdescription: string;
  //user: UserModel;

  static cloneBase(project: ProjectModel): ProjectModel{
    const cloneProjectModel: ProjectModel = new ProjectModel();
    cloneProjectModel.id = project.id;
    cloneProjectModel.projectdescription = project.projectdescription;
    cloneProjectModel.projectcode = project.projectcode;
    //cloneProjectModel.user = project.user;
    return cloneProjectModel;
  }
}

File: fapi/src/main/java/com/netcracker/edu/project/fapi/model/ProjectModel.java
package com.netcracker.edu.project.fapi.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@JsonIgnoreProperties(ignoreUnknown = true)
public class ProjectModel {
    private int id;
    private String projectcode;
    private String projectdescription;


    public ProjectModel() {
    }

    public ProjectModel(int id, String projectcode, String projectdescription) {
        this.id = id;
        this.projectcode = projectcode;
        this.projectdescription = projectdescription;


    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getProjectdescription() {
        return projectdescription;
    }

    public void setProjectdescription(String projectdescription) {
        this.projectdescription = projectdescription;
    }

    public String getProjectcode() {
        return projectcode;
    }

    public void setProjectcode(String projectcode) {
        this.projectcode = projectcode;
    }

}
